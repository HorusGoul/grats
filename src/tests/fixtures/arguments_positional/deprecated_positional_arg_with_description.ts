/** @gqlType */
class User {
  /** @gqlField */
  greeting(
    /**
     * A great name
     * @deprecated
     */
    name?: string | null,
  ): string {
    return `Hello ${name}`;
  }
}
