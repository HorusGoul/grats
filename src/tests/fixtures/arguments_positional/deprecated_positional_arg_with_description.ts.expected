-----------------
INPUT
----------------- 
/** @gqlType */
class User {
  /** @gqlField */
  greeting(
    /**
     * A great name
     * @deprecated
     */
    name?: string | null,
  ): string {
    return `Hello ${name}`;
  }
}

-----------------
OUTPUT
-----------------
-- SDL --
type User {
  greeting(
    """A great name"""
    name: String @deprecated @positionalArg(position: 0)
  ): String
}
-- TypeScript --
import { GraphQLSchema, GraphQLObjectType, GraphQLString } from "graphql";
export function getSchema(): GraphQLSchema {
    const UserType: GraphQLObjectType = new GraphQLObjectType({
        name: "User",
        fields() {
            return {
                greeting: {
                    name: "greeting",
                    type: GraphQLString,
                    args: {
                        name: {
                            description: "A great name",
                            deprecationReason: "No longer supported",
                            name: "name",
                            type: GraphQLString
                        }
                    },
                    resolve(source, args, context, info) {
                        return typeof source.resolve === "function" ? source.resolve(args.name) : source.resolve;
                    }
                }
            };
        }
    });
    return new GraphQLSchema({
        types: [UserType]
    });
}
